src/components/attach/AttachHub.js

```diff
@@ -0,0 +1,406 @@
+import AttachmentItem from "@components/attach/components/AttachmentItem";
+import CustomScrollBar from "@components/_helpers/CustomScrollBar";
+import DownloadManager from "@src/adapters/downloadManager";
+import OvalLoader from "@components/_helpers/OvalLoader";
+import api from "@api/api";
+import compressFile from "@utils/media/compress_file";
+import encodeImageToBlurhash from "@utils/media/get_blur_hash";
+import globalConstants from "@helpers/constants";
+import heicToPng from "@utils/media/heic_to_png";
+import removeAndNavigateLastSection from "@utils/navigation/get_prev_page";
+import showCustomAlert from "@utils/show_alert";
+import TextAreaInput from "@components/hub/elements/TextAreaInput";
+import { KEY_CODES } from "@helpers/keyCodes";
+import {
+  addMessage,
+  removeMessage,
+  selectAllMessages,
+} from "@store/values/Messages";
+import {
+  getConverastionById,
+  setLastMessageField,
+  updateLastMessageField,
+} from "@store/values/Conversations";
+import { getIsMobileView } from "@store/values/IsMobileView";
+import { getNetworkState } from "@store/values/NetworkState";
+import { selectCurrentUser } from "@store/values/CurrentUser";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useLocation } from "react-router-dom";
+
+import "@styles/attach/AttachHub.css";
+
+export default function AttachHub() {
+  const dispatch = useDispatch();
+  const { pathname, hash } = useLocation();
+
+  const connectState = useSelector(getNetworkState);
+  const isMobile = useSelector(getIsMobileView);
+
+  const messages = useSelector(selectAllMessages);
+  const selectedConversation = useSelector(getConverastionById);
+  const selectedCID = selectedConversation._id;
+  const currentUser = useSelector(selectCurrentUser);
+
+  const [isPending, setIsPending] = useState(false);
+  const [isSendMessageDisable, setIsSendMessageDisable] = useState(false);
+
+  const inputFilesRef = useRef(null);
+  const inputTextRef = useRef(null);
+
+  const [files, setFiles] = useState([]);
+
+  const addFiles = async (newFiles) => {
+    if (!newFiles.length) {
+      return;
+    }
+
+    setIsPending(true);
+
+    async function compressAndHashFile(file) {
+      file = await compressFile(file);
+      const localFileUrl = URL.createObjectURL(file);
+      file.localUrl = localFileUrl;
+
+      try {
+        file.blurHash = await encodeImageToBlurhash(localFileUrl);
+      } catch (e) {
+        file.blurHash = globalConstants.defaultBlurHash;
+      }
+
+      return file;
+    }
+
+    const selectedFiles = [];
+    try {
+      for (let i = 0; i < newFiles.length; i++) {
```

Comment:
maybe we also could extract that boilerplate into some util or so, so the AttachHub component will be smaller and easier to read